src/main/proto/logon.proto

```diff
@@ -7,9 +7,18 @@ option java_package = "com.Elessar.proto";
 message LogonRequest {
   required string name = 1;
   required string password = 2;
+  required string clientURL = 3;
 }
 
 message LogonResponse {
   required bool success = 1;
   optional string failReason = 2;
-}
\ No newline at end of file
+  repeated UnreadMsg messages = 3;
```

Comment:
as we discussed, we don't need the unread message in LogOnResponse anymore. Instead, we need a new request sending all undelivered message from server to the client